src/service/analytics.service.ts

```diff
@@ -0,0 +1,154 @@
+
+// Default values for initializing the analytics client.
+const DEVELOPMENT_KEY = '0e26030f769c1e630c59e1b3dec37957';
+const PRODUCTION_KEY = '13c1581286213207b29bc7fc47e787e7';
+const HOST_NAME_REGEX = new RegExp('localhost');
+const PRODUCT_NAME = 'bb_help_widget';
+const DEFAULT_CONFIG = {
+  persistence: 'localStorage',
+  protocol: 'https'
+};
+
+export class AnalyticsService {
+  public jwtDecoder: any;
+  private superProperties: any;
+  private analyticsClient: any;
+  private decodedToken: any;
+  private configService: ConfigService;
+
+  constructor() {
+    this.decodedToken = {};
+    this.configService = new ConfigService();
+    this.setSuperProperties({
+      'Referring Service Name': this.configService.getConfigAttribute('productId')
+    });
+  }
+
+  public setupMixpanel() {
+    this.setAnalyticsClient(this.getMixpanel());
+    this.jwtDecoder = this.getJwtDecoder();
+    this.initMixpanel();
+    this.setupAnalyticsClient();
+  }
+
+  public trackEvent(eventName: string, payload: any) {
+    console.log(eventName);
+    Object.keys(payload)
+      .forEach((property) => {
+        const titleCasePropertyName = this.camelToTitleCase(property);
+        if (titleCasePropertyName !== property) {
+          payload[titleCasePropertyName] = payload[property];
+          delete payload[property];
+        }
+      });
+
+    console.log('client', this.getAnalyticsClient().bb_help_widget);
+    this.getAnalyticsClient().bb_help_widget.track(eventName, payload);
+  }
+
+  private getSuperProperties() {
+    return this.superProperties;
+  }
+
+  private setSuperProperties(props: any) {
+    this.superProperties = props;
+  }
+
+  private getAnalyticsClient() {
+    return this.analyticsClient;
+  }
+
+  private setAnalyticsClient(client: any) {
+    this.analyticsClient = client;
+  }
+
+  private getDecodedToken() {
+    return this.decodedToken;
+  }
+
+  private setDecodedToken(token: any) {
+    this.decodedToken = token;
+  }
+
+  // Converts 'pageName' to 'Page Name', and doesn't mangle properties like 'Duration (s)'.
+  private camelToTitleCase(oldString: string) {
+    return oldString.replace(/ /g, '')
+      .replace(CAMEL_TO_TITLE_CASE_REGEX, '$1 ')
+      .replace(/^./, (match) => {
+        return match.toUpperCase();
+      });
+  }
+
+  private getMixpanel() {
+    return require('mixpanel-browser');
+  }
+
+  private getJwtDecoder() {
+    return require('jwt-decode');
+  }
+
+  private initMixpanel() {
+    const MIXPANEL_KEY = this.getMixpanelKey();
+    this.getAnalyticsClient().init(MIXPANEL_KEY, DEFAULT_CONFIG, PRODUCT_NAME);
+  }
+
+  private getMixpanelKey() {
+    if (this.isDevelopment()) {
+      return DEVELOPMENT_KEY;
+    }
+    return PRODUCTION_KEY;
+  }
+
+  private isDevelopment() {
+    // Converts the returned value to a boolean. If the hostname matches localhost, return true.
+    return !!window.location.hostname.match(HOST_NAME_REGEX);
```

Comment:
We should meet with @Blackbaud-BobbyEarl, and see if there is a way we can determine if the SKYUX app is also in development mode.  checking for localhost will work for regular JS apps running localhost, but won't account for dev environments or host.nxt running locally.

Reply:
```
Bobby Earl [1:41 PM]
If in dev, there’s a `SKY_PAGES_DEV_INFO` global variable that exists.
```
We should add this check to make sure we are using Dev for skyux 2 dev mode apps.  

` && window.SKY_PAGES_DEV_INTO !== undefined`; or something.